stock_account_move_multi_reset_to_draft/models/account_move.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 Tecnativa - Víctor Martínez
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import models
+from odoo.tools.float_utils import float_is_zero
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def button_draft(self):
+        """If it is a purchase invoice, we will create a new SVL for each line with
+        the sum of the value in opposite sign.
+        """
+        res = super().button_draft()
+        for item in self.sudo().filtered(
+            lambda x: x.is_inbound
+            and any(line.stock_valuation_layer_ids for line in x.line_ids)
+        ):
+            for line in item.line_ids.filtered("stock_valuation_layer_ids"):
+                svls = line.stock_valuation_layer_ids
+                value = sum(svls.mapped("value"))
+                if not float_is_zero(
+                    value, precision_rounding=line.currency_id.rounding
+                ):
+                    svls[0].copy({"value": -value})
+        return res
```

Comment:
I am trying to add this change and creating a test with the use case you describe but there are several considerations:
- I failed to reproduce the error in the step “_Reset the first bill to draft -> User error to prevent valuation inconsistencies_”.
- I think that the `_validate_accounting_entries()` method should not be called.
- I think changing the `standard_price` is something this module should not do.

In general I think it is adding too much logic to something that should not have it, you should be able to change to draft as many times as you want (at least that's my opinion).

Reply:
I have added all the changes you indicate, but in the test this step _Reset the first bill to draft -> User error to prevent valuation inconsistencies_ does not work.